src/test/java/com/baeldung/selenium/alertsandpopups/HandleAlertsUnitTests.java

```diff
@@ -0,0 +1,140 @@
+        driver.quit();
+    }
+
+    @Test
+    public void whenAlertIsNotHandled_thenThrowException() {
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://testpages.herokuapp.com/styled/alerts/alert-test.html");
+
+        driver.findElement(By.id("alertexamples"))
+            .click();
+
+        assertThrows(UnhandledAlertException.class, () -> {
+            driver.findElement(By.xpath("/html/body/div/div[1]/div[2]/a[2]"))
+                .click();
+        });
+
+        driver.quit();
+    }
+
+    @Test
+    public void whenConfirmationAlertIsTriggered_thenHandleIt() {
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://testpages.herokuapp.com/styled/alerts/alert-test.html");
+        driver.findElement(By.id("confirmexample"))
+            .click();
+
+        Alert alert = driver.switchTo()
+            .alert();
+        String alertText = alert.getText();
+        alert.accept();
+
+        assertEquals("true", driver.findElement(By.id("confirmreturn"))
+            .getText());
+
+        driver.findElement(By.id("confirmexample"))
+            .click();
+
+        alert = driver.switchTo()
+            .alert();
+        alert.dismiss();
+    
+        assertEquals("false", driver.findElement(By.id("confirmreturn"))
+            .getText());
+        driver.quit();
+    }
+
+    @Test
+    public void whenPromptAlertIsTriggered_thenHandleIt() {
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://testpages.herokuapp.com/styled/alerts/alert-test.html");
+
+        driver.findElement(By.id("promptexample"))
+            .click();
+
+        Alert alert = driver.switchTo()
+            .alert();
+
+        String inputText = "Selenium Test";
+        alert.sendKeys(inputText);
+
+        alert.accept();
+
+        assertEquals(inputText, driver.findElement(By.id("promptreturn"))
+            .getText());
+
+        driver.quit();
+    }
+
+    @Test
+    public void whenAlertIsPresent_thenWaitAndHandle() {
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://testpages.herokuapp.com/styled/alerts/alert-test.html");
+
+        driver.findElement(By.id("alertexamples"))
+            .click();
+
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        Alert alert = wait.until(ExpectedConditions.alertIsPresent());
+
+        String alertText = alert.getText();
+        alert.accept();
+
+        assertEquals("I am an alert box!", alertText);
+
+        driver.quit();
+    }
+
+
+    @Test
+    public void whenNoAlertIsPresent_thenHandleGracefully() {
+        WebDriver driver = new ChromeDriver();
+        driver.get("https://testpages.herokuapp.com/styled/alerts/alert-test.html");
+
+        try {
+            Alert alert = driver.switchTo()
+                .alert();
+            alert.accept();
+        } catch (NoAlertPresentException e) {
+
+        }
```

Comment:
Not resolved: We aren't testing that this is actually thrown - if it's not thrown, the test is still green